### PR TITLE
CI/CD: Release trigger

### PR DESCRIPTION
**Pull request by JacobMGEvans**
### Tests run before other workflows
To be sure that Tests, Linting and type checking is occurring on code before it is merged and released/pre-released other workflows have the Testing workflow as a trigger. For example the Tests need to complete and be successful before the prerelease will run.
### Flow tested on fork
Tested the full contribution -> changeset -> PR -> merge -> prerelease and release flow on a fork, to as closely test the GitHub Workflows are properly running in a realistic situation. 
### Release flow triggered on PR's closed
Whenever the a PR is closed/merged the Workflow will trigger for checking for a changeset and adding it to the changeset "Version" PR, when the "Version" PR is closed/merged it will trigger a release.

### CODE REVIEW

**Comment by threepointone** [comment]
Could you please explain what this PR does? The description doesn't explain much.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,66 +1,47 @@
-
 name: Release
 
 on:
   pull_request:
     types: [closed]
-    branches:
-      - changeset-release/main
```
Comment (threepointone) [inline .github/workflows/release.yml:8]: add a target branch main

**Review by threepointone — APPROVED** [review]
alright, let's try this out

File: .github/workflows/release.yml (hunk 1 above)
Comment (JacobMGEvans) [reply .github/workflows/release.yml:8]: Will do!

**Comment by threepointone** [comment]
I think this might have broken something https://github.com/cloudflare/wrangler2/runs/4558536619?check_suite_focus=true

**Pull request merged**